sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1594,6 +1594,15 @@ object SQLConf {
         "WHERE, which does not follow SQL standard.")
       .booleanConf
       .createWithDefault(false)
+
+  val ALIAS_NON_STRUCT_GROUPING_KEY_AS_VALUE =
+    buildConf("spark.sql.legacy.dataset.aliasNonStructGroupingKeyAsValue")
+      .internal()
+      .doc("When set to true, the key attribute resulted from running `Dataset.groupByKey` " +
+        "for non-struct key type, will be named as `value`, following the behavior of Spark " +
```

Comment:
so it's "named" as value, not "alias" to value. I think it's more precise to call this config `spark.sql.legacy.dataset.nameNonStructGroupingKeyAsValue`?

Reply:
`nameNonStructGroupingKeyAsValue` is fine to me.